### DISCUSSION TITLE
getting error while configuring hubs cloud personal

### DISCUSSION THREAD

**Discussion by RBOSAMIYA9** [mozilla/hubs#4803] [Hubs Cloud Questions] [answered]
Hello all i am getting an error while configuring hubs cloud personal into aws

![image](https://user-images.githubusercontent.com/27606753/140062544-6b533f9b-a774-444f-bc05-7010bc66f197.png)

further digging bottom most error I have found

![image](https://user-images.githubusercontent.com/27606753/140062631-50ff5501-7802-402b-984d-b095d4b02eee.png)

error: 
`Resource handler returned message: "The runtime parameter of nodejs10.x is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (nodejs14.x) while creating or updating functions. (Service: Lambda, Status Code: 400, Request ID: 11a94f55-f241-4e4b-a27f-6a87069f583e, Extended Request ID: null)" (RequestToken: 68b2a555-2b30-a966-5fe5-a5efba971a06, HandlerErrorCode: InvalidRequest)`

I have also checked into stack template but i haven't found any traces of node10.x 
so from where this template is coming from and How can i edit this template to solve the error

Or is there any other way to solve this problem.

**Comment by pattersonbl2** [comment]
Can you send me a screen shot of the template you are using ?  The cloud formation has lambda runtimes that are reflecting 10x but we have since updated this months ago.  We are currently still having issues with this for another runtime and working with amazon to upload a correction.

**Reply by bana118** [reply]
I also get the same error and deployment fails.
The template I am using is the following file (I changed the file extension from .template to .txt).
[5f452d36-bfff-4c02-a417-5c663b64fd08.fa1a8724-998c-410c-9d58-e8b7fe698727.txt](https://github.com/mozilla/hubs/files/7486477/5f452d36-bfff-4c02-a417-5c663b64fd08.fa1a8724-998c-410c-9d58-e8b7fe698727.txt)

I tried replacing `nodejs10.x` with `node14.x`, but I got the following error.

```
Embedded stack arn:aws:cloudformation:ap-northeast-1:671184214334:stack/vrdevel-hubs-NearsparkApplication-EKKXJZABU78I/0a8fc9e0-3e3f-11ec-80c0-0616a1c70b31 was not successfully created: The following resource(s) failed to create: [ThumbnailFunction].
```

**Reply by pattersonbl2** [reply]
You may need to change this file to a .yaml file.   The error is related to the stack not being able to create a nested CFT stack.   Sorry for the delay we have been having issues with CFT update process with AWS

**Reply by bana118** [reply]
The template update seems to have been reflected and node.js 14.x is now used.
However, the following error occurs.

![stack](https://user-images.githubusercontent.com/43720583/140615500-2df7f212-8313-447a-a531-5e0244f8beab.png)

This is the same issue as in the discussion https://github.com/mozilla/hubs/discussions/4805.

**Reply by RBOSAMIYA9** [reply]
I guess the error occurs because, main template is updated but sub templates for creating the sub stacks are not updates, I have also tried to create change sets and then retry the process but I was not able to retry.


References:

**Main template s3 link**: https://s3.amazonaws.com/awsmp-fulfillment-cf-templates-prod/5f452d36-bfff-4c02-a417-5c663b64fd08.fa1a8724-998c-410c-9d58-e8b7fe698727.template


### error causing sub templates
Following is one of error error causing template. There are more such error causing templates having **runtime: nodejs10.x**
**KeymasterApplication**
`{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Outputs": {
    "GenerateUserKeyFunctionArn": {
      "Description": "Keymaster GenerateUserKeyFunction ARN",
      "Value": {
        "Fn::GetAtt": [
          "GenerateUserKeyFunction",
          "Arn"
        ]
      }
    },
    "GenerateServiceKeyFunctionArn": {
      "Description": "Keymaster GenerateServiceKeyFunction ARN",
      "Value": {
        "Fn::GetAtt": [
          "GenerateServiceKeyFunction",
          "Arn"
        ]
      }
    },
    "GenerateVapidKeysFunctionArn": {
      "Description": "Keymaster GenerateVapidKeysFunction ARN",
      "Value": {
        "Fn::GetAtt": [
          "GenerateVapidKeysFunction",
          "Arn"
        ]
      }
    },
    "GenerateSecretFunctionArn": {
      "Description": "Keymaster GenerateSecretFunction ARN",
      "Value": {
        "Fn::GetAtt": [
          "GenerateSecretFunction",
          "Arn"
        ]
      }
    },
    "GenerateTOTPFunctionArn": {
      "Description": "Keymaster GenerateTOTPFunction ARN",
      "Value": {
        "Fn::GetAtt": [
          "GenerateTOTPFunction",
          "Arn"
        ]
      }
    },
    "GenerateJWTKeysFunctionArn": {
      "Description": "Keymaster GenerateJWTKeysFunction ARN",
      "Value": {
        "Fn::GetAtt": [
          "GenerateJWTKeysFunction",
          "Arn"
        ]
      }
    },
    "GenerateRingKeyFunctionArn": {
      "Description": "Keymaster GenerateRingKeyFunction ARN",
      "Value": {
        "Fn::GetAtt": [
          "GenerateRingKeyFunction",
          "Arn"
        ]
      }
    },
    "ReadSecretFunctionArn": {
      "Description": "Keymaster ReadSecretFunction ARN",
      "Value": {
        "Fn::GetAtt": [
          "ReadSecretFunction",
          "Arn"
        ]
      }
    },
    "DeleteParametersFunctionArn": {
      "Description": "Keymaster DeleteParametersFunction ARN",
      "Value": {
        "Fn::GetAtt": [
          "DeleteParametersFunction",
          "Arn"
        ]
      }
    }
  },
  "Resources": {
    "GenerateTOTPFunction": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "S3Bucket": "awsserverlessrepo-changesets-plntc6bfnfj",
          "S3Key": "724323103689/arn:aws:serverlessrepo:us-east-1:771996922080:applications-keymaster-public-5f452d36-bfff-4c02-a417-5c663b64fd08-versions-0.1.13/6c019460-cda5-45ca-8740-082ee2c57ab3"
        },
        "Tags": [
          {
            "Value": "SAM",
            "Key": "lambda:createdBy"
          }
        ],
        "MemorySize": 512,
        "Handler": "index.generate_totp",
        "Role": {
          "Ref": "KeymasterRoleArn"
        },
        "Timeout": 30,
        "Runtime": "nodejs10.x"
      }
    },
    "GenerateVapidKeysFunction": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "S3Bucket": "awsserverlessrepo-changesets-plntc6bfnfj",
          "S3Key": "724323103689/arn:aws:serverlessrepo:us-east-1:771996922080:applications-keymaster-public-5f452d36-bfff-4c02-a417-5c663b64fd08-versions-0.1.13/6c019460-cda5-45ca-8740-082ee2c57ab3"
        },
        "Tags": [
          {
            "Value": "SAM",
            "Key": "lambda:createdBy"
          }
        ],
        "MemorySize": 512,
        "Handler": "index.generate_vapid_keys",
        "Role": {
          "Ref": "KeymasterRoleArn"
        },
        "Timeout": 30,
        "Runtime": "nodejs10.x"
      }
    },
    "GenerateServiceKeyFunction": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "S3Bucket": "awsserverlessrepo-changesets-plntc6bfnfj",
          "S3Key": "724323103689/arn:aws:serverlessrepo:us-east-1:771996922080:applications-keymaster-public-5f452d36-bfff-4c02-a417-5c663b64fd08-versions-0.1.13/6c019460-cda5-45ca-8740-082ee2c57ab3"
        },
        "Tags": [
          {
            "Value": "SAM",
            "Key": "lambda:createdBy"
          }
        ],
        "MemorySize": 512,
        "Handler": "index.generate_service_key",
        "Role": {
          "Ref": "KeymasterRoleArn"
        },
        "Timeout": 30,
        "Runtime": "nodejs10.x"
      }
    },
    "GenerateRingKeyFunction": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "S3Bucket": "awsserverlessrepo-changesets-plntc6bfnfj",
          "S3Key": "724323103689/arn:aws:serverlessrepo:us-east-1:771996922080:applications-keymaster-public-5f452d36-bfff-4c02-a417-5c663b64fd08-versions-0.1.13/6c019460-cda5-45ca-8740-082ee2c57ab3"
        },
        "Tags": [
          {
            "Value": "SAM",
            "Key": "lambda:createdBy"
          }
        ],
        "MemorySize": 512,
        "Handler": "index.generate_ring_key",
        "Role": {
          "Ref": "KeymasterRoleArn"
        },
        "Timeout": 30,
        "Runtime": "nodejs10.x"
      }
    },
    "DeleteParametersFunction": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "S3Bucket": "awsserverlessrepo-changesets-plntc6bfnfj",
          "S3Key": "724323103689/arn:aws:serverlessrepo:us-east-1:771996922080:applications-keymaster-public-5f452d36-bfff-4c02-a417-5c663b64fd08-versions-0.1.13/6c019460-cda5-45ca-8740-082ee2c57ab3"
        },
        "Tags": [
          {
            "Value": "SAM",
            "Key": "lambda:createdBy"
          }
        ],
        "MemorySize": 512,
        "Handler": "index.delete_parameters",
        "Role": {
          "Ref": "KeymasterRoleArn"
        },
        "Timeout": 30,
        "Runtime": "nodejs10.x"
      }
    },
    "GenerateSecretFunction": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "S3Bucket": "awsserverlessrepo-changesets-plntc6bfnfj",
          "S3Key": "724323103689/arn:aws:serverlessrepo:us-east-1:771996922080:applications-keymaster-public-5f452d36-bfff-4c02-a417-5c663b64fd08-versions-0.1.13/6c019460-cda5-45ca-8740-082ee2c57ab3"
        },
        "Tags": [
          {
            "Value": "SAM",
            "Key": "lambda:createdBy"
          }
        ],
        "MemorySize": 512,
        "Handler": "index.generate_secret",
        "Role": {
          "Ref": "KeymasterRoleArn"
        },
        "Timeout": 30,
        **"Runtime": "nodejs10.x"**
      }
    },
    "ReadSecretFunction": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "S3Bucket": "awsserverlessrepo-changesets-plntc6bfnfj",
          "S3Key": "724323103689/arn:aws:serverlessrepo:us-east-1:771996922080:applications-keymaster-public-5f452d36-bfff-4c02-a417-5c663b64fd08-versions-0.1.13/6c019460-cda5-45ca-8740-082ee2c57ab3"
        },
        "Tags": [
          {
            "Value": "SAM",
            "Key": "lambda:createdBy"
          }
        ],
        "MemorySize": 512,
        "Handler": "index.read_secret",
        "Role": {
          "Ref": "KeymasterRoleArn"
        },
        "Timeout": 30,
        "Runtime": "nodejs10.x"
      }
    },
    "GenerateJWTKeysFunction": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "S3Bucket": "awsserverlessrepo-changesets-plntc6bfnfj",
          "S3Key": "724323103689/arn:aws:serverlessrepo:us-east-1:771996922080:applications-keymaster-public-5f452d36-bfff-4c02-a417-5c663b64fd08-versions-0.1.13/6c019460-cda5-45ca-8740-082ee2c57ab3"
        },
        "Tags": [
          {
            "Value": "SAM",
            "Key": "lambda:createdBy"
          }
        ],
        "MemorySize": 512,
        "Handler": "index.generate_jwt_keys",
        "Role": {
          "Ref": "KeymasterRoleArn"
        },
        "Timeout": 30,
        "Runtime": "nodejs10.x"
      }
    },
    "GenerateUserKeyFunction": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "S3Bucket": "awsserverlessrepo-changesets-plntc6bfnfj",
          "S3Key": "724323103689/arn:aws:serverlessrepo:us-east-1:771996922080:applications-keymaster-public-5f452d36-bfff-4c02-a417-5c663b64fd08-versions-0.1.13/6c019460-cda5-45ca-8740-082ee2c57ab3"
        },
        "Tags": [
          {
            "Value": "SAM",
            "Key": "lambda:createdBy"
          }
        ],
        "MemorySize": 512,
        "Handler": "index.generate_user_key",
        "Role": {
          "Ref": "KeymasterRoleArn"
        },
        "Timeout": 30,
        "Runtime": "nodejs10.x"
      }
    }
  },
  "Description": "Keymaster\n",
  "Parameters": {
    "KeymasterRoleArn": {
      "Type": "String"
    }
  }
}`

###  error causing templates names
- KeymasterApplication
- SpeelycaptorApplication
- PhotomnemonicApplication
- NearsparkApplication

**Reply by bana118** [reply]
I tried it today and was able to deploy it. It seems that the template update has been reflected.  Thank you!

**Comment by pattersonbl2** [comment, marked as answer]
The marketplace CFT has been updated. Feel free to test.
